### PR TITLE
IGNITE-12155: [ML] Implementation of distributed estimator

### PR DESCRIPTION
**Pull request by avplatonov**

### CODE REVIEW

File: examples/src/main/java/org/apache/ignite/examples/ml/tutorial/hyperparametertuning/Step_12_Parallel_Random_Search.java [hunk 1]
```diff
@@ -133,11 +133,7 @@ public static void main(String[] args) {
                     .addHyperParam("p", normalizationTrainer::withP, new Double[]{1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0, 10.0})
                     .addHyperParam("maxDeep", trainerCV::withMaxDeep, new Double[]{1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0, 10.0})
                     .addHyperParam("minImpurityDecrease", trainerCV::withMinImpurityDecrease, new Double[]{0.0, 0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1.0});
-                
-                BinaryClassificationMetrics metrics = (BinaryClassificationMetrics) new BinaryClassificationMetrics()
-                    .withNegativeClsLb(0.0)
-                    .withPositiveClsLb(1.0)
-                    .withMetric(BinaryClassificationMetricValues::accuracy);
+
```
Comment (zaleslaw) [inline examples/src/main/java/org/apache/ignite/examples/ml/tutorial/hyperparametertuning/Step_12_Parallel_Random_Search.java]: remove empty line, please

File: modules/ml/src/main/java/org/apache/ignite/ml/dataset/impl/cache/CacheBasedDatasetBuilder.java [hunk 2]
```diff
@@ -33,6 +31,9 @@
 import org.apache.ignite.ml.environment.LearningEnvironment;
 import org.apache.ignite.ml.environment.LearningEnvironmentBuilder;
 
+import java.io.Serializable;
```
Comment (zaleslaw) [inline modules/ml/src/main/java/org/apache/ignite/ml/dataset/impl/cache/CacheBasedDatasetBuilder.java]: Something wrong with order of imports, not in Ignite codestyle

File: modules/ml/src/main/java/org/apache/ignite/ml/selection/cv/AbstractCrossValidation.java [hunk 3]
```diff
@@ -39,21 +28,23 @@
...
 import org.apache.ignite.ml.selection.scoring.metric.Metric;
+import org.apache.ignite.ml.selection.scoring.metric.MetricName;
 import org.apache.ignite.ml.selection.split.mapper.SHA256UniformMapper;
 import org.apache.ignite.ml.selection.split.mapper.UniformMapper;
 import org.apache.ignite.ml.trainers.DatasetTrainer;
 import org.apache.ignite.ml.util.genetic.Chromosome;
 import org.apache.ignite.ml.util.genetic.GeneticAlgorithm;
 import org.jetbrains.annotations.NotNull;
 
+import java.util.*;
```
Comment (zaleslaw) [inline modules/ml/src/main/java/org/apache/ignite/ml/selection/cv/AbstractCrossValidation.java]: incorrect order of imports

File: modules/ml/src/main/java/org/apache/ignite/ml/selection/scoring/evaluator/Evaluator.java [hunk 4]
```diff
@@ -41,358 +56,585 @@
...
      * @param preprocessor The preprocessor.
-     * @param metric       The binary classification metric.
-     * @param <K>          The type of cache entry key.
-     * @param <V>          The type of cache entry value.
+     * @param metric The metric.
+     * @param <K> The type of cache entry key.
+     * @param <V> The type of cache entry value.
      * @return Computed metric.
      */
-    public static <L, K, V> double evaluate(IgniteCache<K, V> dataCache,
```
Comment (zaleslaw) [inline modules/ml/src/main/java/org/apache/ignite/ml/selection/scoring/evaluator/Evaluator.java]: Use Ignite codestyle for formatting

**Comment by zaleslaw** [comment]
I suggest to use Ignite codestyle for formatting code and import order

**Comment by avplatonov** [comment]
@zaleslaw Thanks, I set up ignite style in my IDE (idea) according to https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines and reformat metrics code and examples.

**Pull request merged**